docs/explanations/architecture-2.md

```diff
@@ -0,0 +1,239 @@
+and downloads all the Kubewarden policies specified.
+You can download Kubewarden policies from remote HTTP servers and container registries.
+
+You use policy settings parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
+
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[specification reference](../reference/spec/01-intro-spec.md)
+section of the documentation.
+
+The `policy-server` exits with an error
+if one or more policies received wrong configuration parameters
+from the policy specification provided by the user.
+
+If all policies are correctly configured,
+`policy-server`
+spawns a pool of worker threads
+to evaluate incoming requests
+using the Kubewarden policies
+specified by the user.
+
+Finally, the `policy-server` starts a HTTPS server,
+the Kubewarden Policy Server,
+listening to incoming validation requests.
+Kubewarden uses the TLS key and certificate
+created by `kubewarden-controller`
+to secure the web server.
+
+The web server exposes each policy by a dedicated path
+following the naming convention: `/validate/<policy ID>`.
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
+
+### Making Kubernetes aware of the policy
+
+The `policy-server` Pods have a
+[`Readiness Probe`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/),
+that `kubewarden-controller` uses to check when
+the `policy-server` Deployment is ready to evaluate [`AdmissionReview`](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response)s.
+
+Once the `policy-server` Deployment is marked as `Ready`,
+the `kubewarden-controller` makes the Kubernetes API server
+aware of the new policy by creating either a
+`MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration` object.
+
+Each policy has a dedicated
+`MutatingWebhookConfiguration` or `ValidatingWebhookConfiguration`
+which points to the Webhook endpoint served by `policy-server`.
+The endpoint is reachable by the `/validate/<policy ID>` URL.
+
+### Policy in action
+
+Now that all the necessary plumbing is complete,
+Kubernetes starts sending Admission Review requests to the right `policy-server` endpoints.
+
+A `policy-server` receives the Admission Request object and,
+based on the endpoint that received the request,
+uses the correct policy to evaluate it.
+
+Each policy is evaluated inside its own dedicated WebAssembly sandbox.
+The communication between `policy-server` (the "host")
+and the WebAssembly policy (the "guest")
+uses the waPC communication protocol.
+The protocol description is part of the [writing policies](../tutorials/writing-policies/index.md) documentation.
```

Comment:
Maybe is worth to mention the [WASI](https://docs.kubewarden.io/tutorials/writing-policies/wasi/intro-wasi) policies as well